cli/main.go

```diff
@@ -154,42 +152,45 @@ func fsDaemon(
 	logger.Printf("building config")
 
 	// Fuse mount
+	// MountConfig https://github.com/jacobsa/fuse/blob/3e9d24d5e3ffe84da952c40fc8d3f42df661466d/mount_config.go#L215
 	cfg := &fuse.MountConfig{
 		FSName:      "dxfuse",
 		ErrorLogger: logger,
 		DebugLogger: fuse_logger,
 
 		// Required for sequential writes
 		DisableWritebackCaching: true,
-		Options:                 mountOptions,
+		// Pass read-only mount option if not in limitedWrite
+		ReadOnly: options.ReadOnly,
```

Comment:
https://github.com/jacobsa/fuse/blob/3e9d24d5e3ffe84da952c40fc8d3f42df661466d/mount_config.go#L258-L261

Instead of setting `mountOptions["ro"] = ""` in the additional mountOptions we can use the param `ReadOnly` that is already included int the MountConfig class